src/conv-core/isomalloc.C

```diff
@@ -664,233 +727,126 @@ static void CmiIsomallocInitExtent()
       memRange_t start = CMIALIGN((uintptr_t)freeRegion.start, division_size);
       memRange_t end = CMIALIGN((uintptr_t)freeRegion.start + freeRegion.len - (division_size-1), division_size);
       memRange_t len = end - start;
-      freeRegion.start = (uint8_t *)start;
-      freeRegion.len = len;
+      IsoRegion.s = start;
+      IsoRegion.e = end;
       DEBUG_PRINT("[%d] Isomalloc memory region: 0x%zx - 0x%zx (%zu gigs)\n", CmiMyPe(),
                   start, end, len/gig);
     }
-  } /* end if myrank == 0 */
+  }
 
-  CmiNodeAllBarrier();
+  /* isomalloc_sync
+   * Available address space regions can vary across nodes.
+   * Calculate the intersection of all memory regions on all nodes.
+   */
 
-  /*
-     on some machines, isomalloc memory regions on different nodes
-     can be different. use +isomalloc_sync to calculate the
-     intersect of all memory regions on all nodes.
-     */
-  if (_sync_iso == 1)
+  CmiAssignOnce(&CmiIsomallocSyncHandlerNode0Idx, CmiRegisterHandler(CmiIsomallocSyncHandlerNode0));
+  CmiAssignOnce(&CmiIsomallocSyncHandlerOtherNodesIdx, CmiRegisterHandler(CmiIsomallocSyncHandlerOtherNodes));
+
+#if __FAULT__
+  if (CmiIsomallocRestart)
   {
-#ifdef __FAULT__
-    if (_restart == 1)
-    {
-      CmiUInt8 s = (CmiUInt8)freeRegion.start;
-      CmiUInt8 e = (CmiUInt8)(freeRegion.start + freeRegion.len);
-      CmiUInt8 ss, ee;
-      int try_count, fd;
-      char fname[128];
-      sprintf(fname, ".isomalloc");
-      try_count = 0;
-      while ((fd = open(fname, O_RDONLY)) == -1 && try_count < 10000)
-      {
+    if (CmiMyRank() == 0)
+    {
+      CmiAddressSpaceRegion previous;
+
+      int try_count = 0, fd;
+      while ((fd = open(".isomalloc", O_RDONLY)) == -1 && try_count < 10000)
         try_count++;
-      }
+
       if (fd == -1)
-      {
-        CmiAbort(
-            "isomalloc_sync failed during restart, make sure you have a shared file "
-            "system.");
-      }
-      if (read(fd, &ss, sizeof(CmiUInt8)) != sizeof(CmiUInt8))
-      {
-        CmiAbort("Isomalloc> call to read() failed during restart!");
-      }
-      if (read(fd, &ee, sizeof(CmiUInt8)) != sizeof(CmiUInt8))
-      {
+        CmiAbort("isomalloc_sync failed during restart, make sure you have a shared file system.");
+      if (read(fd, &previous, sizeof(CmiAddressSpaceRegion)) != sizeof(CmiAddressSpaceRegion))
         CmiAbort("Isomalloc> call to read() failed during restart!");
-      }
+
       close(fd);
-      if (ss < s || ee > e)
-        CmiAbort(
-            "isomalloc_sync failed during restart, virtual memory regions do not "
-            "overlap.");
+      if (previous.s < IsoRegion.s || previous.e > IsoRegion.e)
+        CmiError("Isomalloc> Warning: virtual memory regions do not overlap: "
```

Comment:
It's weird to have an abort that calls itself a "Warning"

Reply:
It's not an abort, just printing an error message